sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADSeleniumITHelper.java

```diff
@@ -46,6 +49,21 @@ public void logIn() {
         wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='submit']"))).click();
     }
 
+    public boolean logInAADConditionalAccess() {
+        driver.get(app.root() + "oauth2/authorization/azure");
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.name("loginfmt"))).sendKeys(username + Keys.ENTER);
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.name("passwd"))).sendKeys(password + Keys.ENTER);
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='submit']"))).click();
+        driver.get((app.root() + "obo"));
+        try {
+            Thread.sleep(30000);
```

Comment:
why don't we use wait.until here instead of thread sleep?

Reply:
I'm not done yet and will find another way to achieve him.